rust/datafusion/src/catalog/catalog.rs

```diff
@@ -23,6 +23,67 @@ use std::any::Any;
 use std::collections::HashMap;
 use std::sync::{Arc, RwLock};
 
+/// Represent a list of named catalogs
+pub trait CatalogList: Sync + Send {
+    /// Returns the catalog list as [`Any`](std::any::Any)
+    /// so that it can be downcast to a specific implementation.
+    fn as_any(&self) -> &dyn Any;
+
+    /// Adds a new catalog to this catalog list
+    /// If a catalog of the same name existed before, it is replaced in the list and returned.
+    fn register_catalog(
+        &self,
+        name: String,
+        catalog: Arc<dyn CatalogProvider>,
+    ) -> Option<Arc<dyn CatalogProvider>>;
+
+    /// Retrieves the list of available catalog names
+    fn catalog_names(&self) -> Vec<String>;
```

Comment:
Better as `&[&str]` to avoid clone?

Reply:
ah I see in the implementation it is not really possible

Reply:
The way I could think of to avoid the copy is don't make a `CatalogList` trait and instead pass around `MemoryCatalogList` directly. This would allow a function like `fn catalog_names(&self) -> impl IntoIterator<Item=&str>` 

However, given `catalog_names` are only called when creating the information_schema views, I am not sure this optimization is warranted at this time